src/client/opamCommands.ml

```diff
@@ -2424,27 +2429,49 @@ let repository cli =
       let rm = List.filter (fun n -> not (List.mem n names)) in
       let full_wipe = List.mem `All scope in
       let global = global || full_wipe in
-      let gt =
+      let update_gt () =
```

Comment:
What do you think of
```ocaml
      OpamRepositoryState.with_ `Lock_write gt @@ fun rt ->
      if full_wipe then
        let repos =
          OpamRepositoryName.Map.keys rt.OpamStateTypes.repositories
        in
        check_for_repos repos names
          (OpamConsole.warning
             "No configured repositories by these names found: %s");
        OpamRepositoryState.drop @@
        List.fold_left OpamRepositoryCommand.remove rt names
      else begin
        let has_known_repos =
          List.fold_left (fun has_known_repos switch ->
              let repos = OpamRepositoryCommand.switch_repos rt switch in
              check_for_repos' repos names
                (OpamConsole.warning
                   "No configured repositories by these names found in \
                    the selection of switch '%s': %s"
                   (OpamSwitch.to_string switch))
              || has_known_repos)
            false switches
        in
        if scope = [`Current_switch] && has_known_repos then
          OpamConsole.msg
            "Repositories removed from the selections of switch %s. \
             Use '--all' to forget about them altogether.\n"
            (OpamSwitch.to_string (OpamStateConfig.get_switch ()))
      end;
      let rm =
        List.filter (fun n ->
            not (List.exists (OpamRepositoryName.equal n) names))
      in
      ignore @@ OpamRepositoryCommand.update_selection gt
        ~global ~switches:switches rm;
      `Ok ()
```
[diff](https://github.com/ocaml/opam/commit/106c9dfdd529d806af6e8c70b99a9e99e4e05847)
The idea is to factorise some code and make it more step by step, more "readable":
* first loading repostate
* then check availability, display comments, and remove in global case (take advantage of the if)
* then for all, update switch selection (no need to drop as it is in an `with_` function 
* end

Reply:
I incorporated this diff except for the factorization of the repo-state loading as it would load it in write mode regardless of need (non full_wipe do not need it to be in write mode)